libafl/src/observers/stacktrace.rs

```diff
@@ -0,0 +1,66 @@
+//! the ``StacktraceObserver`` looks up the stacktrace on the execution thread and computes a hash for it for dedupe
+
+use ahash::AHasher;
+use core::hash::Hasher;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+use crate::{bolts::tuples::Named, observers::Observer, Error};
+
+/// A simple observer, just overlooking the runtime of the target.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct StacktraceObserver {
+    observer_name: String,
+    hash: Option<u64>,
+}
+
+impl StacktraceObserver {
+    /// Creates a new [`StacktraceObserver`] with the given name.
+    #[must_use]
+    pub fn new(observer_name: String) -> Self {
+        Self {
+            observer_name,
+            hash: None,
+        }
+    }
+
+    /// Gets the runtime for the last execution of this target.
+    #[must_use]
+    pub fn hash(&self) -> &Option<u64> {
+        &self.hash
+    }
+}
+
+impl Default for StacktraceObserver {
+    fn default() -> Self {
+        Self::new("StacktraceObserver".to_string())
+    }
+}
+
+impl<I, S> Observer<I, S> for StacktraceObserver
+where
+    I: Debug,
+{
+    fn pre_exec(&mut self, _state: &mut S, _input: &I) -> Result<(), Error> {
+        Ok(())
+    }
+
+    fn post_exec(&mut self, _state: &mut S, _input: &I) -> Result<(), Error> {
```

Comment:
Reminder for me, here we need an exit_kind to compute the backtrace only on crashes